### PR TITLE
raftstore-v2: reuse failpoint tests in test_early_apply.rs

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #15409

What's Changed:

```commit-message
reuse failpoint tests in test_early_apply
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/test_raftstore-v2/src/util.rs [hunk 1]
```diff
@@ -139,6 +143,13 @@ pub fn configure_for_encryption(config: &mut Config) {
     cfg.master_key = test_util::new_test_file_master_key(manifest_dir);
 }
 
+pub fn configure_for_request_snapshot(config: &mut Config) {
+    // We don't want to generate snapshots due to compact log.
+    config.raft_store.raft_log_gc_threshold = 1000;
+    config.raft_store.raft_log_gc_count_limit = Some(1000);
+    config.raft_store.raft_log_gc_size_limit = Some(ReadableSize::mb(20));
+}
```
Comment (overvenus) [inline components/test_raftstore-v2/src/util.rs:151]: No need duplicate.

Comment (SpadeA-Tang) [reply components/test_raftstore-v2/src/util.rs:151]: why it's duplicate?

Comment (overvenus) [reply components/test_raftstore-v2/src/util.rs:151]: Same as `test_raftstore::configure_for_request_snapshot`.

**Review by overvenus — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Pull request merged**